### ISSUE TITLE
The script is taking long to finish after the last statement is executed.

### ISSUE THREAD

**Issue by sguptatw** [getgauge/taiko#166]
**Expected behavior**
After all the statements are shown as executed, the script execution should finish.

**Actual behavior**
The script is taking long to finish after the last statement is executed.

**Steps to replicate**
* Run the script
```
const { browser, openBrowser, goto, link, click, listItem,checkBox } = require('taiko');

(async () => {
    try {
        await openBrowser();
        await goto("http://todomvc.com/examples/react/#/");
        await write("something");
        await press("Enter");  
        await waitFor(1000);
        await click(checkBox({'class':'toggle'},near('something')))
        await hover(checkBox({'class':'toggle'}))
        await closeBrowser();
    } catch (e) {
        console.error(e);
    } finally {
    }
})();
```
**Version**
```
commit - a9ea9fa69d7676775fc94be0e4c176b746fb7be5
```

**Comment by sguptatw** [comment]
It takes longer in the headful mode. Here is the script to replicate it
```
const { openBrowser, goto, write, press, link, click, near, checkBox, button, reload } = require('taiko');

(async () => {
    try {
        await openBrowser({headless:false});
        await goto("http://todomvc.com/examples/react/#/");
        await write("something");
        await press("Enter");
	    await click(link("Active"));
	    await click(checkBox({'class':'toggle'},near('something')));
	    await click(link("Completed"));
	    await click(button("Clear completed"));
	    await reload("http://todomvc.com/examples/react/#/");
    } catch (e) {
        console.error(e);
    } finally {
        closeBrowser();
    }
})();
```

**Comment by Apoorva-GA** [comment]
Fix should be available in commit 940e180.

**Comment by sguptatw** [comment]
Works as expected on OSx. However taiko is not working on Windows https://github.com/getgauge/taiko/issues/180. 

Todo - verify on Windows
